src/diff/ruleSet.ts

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Rule } from "json-rules-engine";
+import fs from "fs-extra";
+
+/**
+ * Category to define in rule
+ */
+export enum RuleCategory {
+  BREAKING = "Breaking",
+  NON_BREAKING = "Non-Breaking",
+  IGNORED = "Ignored"
+}
+
+/**
+ * Holds rules
+ */
+export class RuleSet {
```

Comment:
```typescript
class RuleSet {
    constructor(public rules: Rule[]) { }
    static async init(rulesPath: string): Promise<RuleSet> {
        return new RuleSet(await fs.readJson(rulesPath));
    }
}
```
The other thread about this suggestion ended up focused on whether to extend Array, but the main point of the suggestion was to use this pattern. Like the `Api` classes, reading from a file should be done in the static method, so that the constructor only has to be concerned with the actual data.